lib/hiera/backend.rb

```diff
@@ -233,6 +249,16 @@ def lookup(key, default, scope, order_override, resolution_type)
       def clear!
         @backends = {}
       end
+
+      def qualified_lookup(segments, hash)
+        value = hash
+        segments.each do |segment|
+          break if value.nil? || value == :undefined
```

Comment:
if it is only in spec tests then that is an indication of unused functionality... will ask R.I